Sources/Jobs/JobQueueHandler.swift

```diff
@@ -68,6 +69,14 @@ final class JobQueueHandler<Queue: JobQueueDriver>: Service {
         var logger = logger
         let startTime = DispatchTime.now().uptimeNanoseconds
 
+        let jobQueuedDuration = Date().timeIntervalSince(queuedJob.queuedAt)
+
+        // Calculate wait time from queued to processing
+        Timer(
+            label: "\(self.metricsLabel)_queued_for_duration_seconds",
+            preferredDisplayUnit: .seconds
+        ).recordNanoseconds(Int64(jobQueuedDuration * 1_000_000_000))
```

Comment:
You are better using the `recordSeconds` function that takes a Double it deals with overflow.

I'm not a great fan of using Doubles as accuracy starts to fail in this situation after about 3 years, but it'd be quite a while longer before the accuracy is so far out it'd be noticeable.